### ISSUE TITLE
Rfc4880KeySelectionStrategy only considers email addresses as uids

### ISSUE THREAD

**Issue by vanitasvitae** [neuhalje/bouncy-gpg#19]
I'm not sure, if this is really an issue on your side, but the Rfc4880KeySelectionStrategy's `publicKeyRingsForUid()` method encloses uid queries with `<` and `>`, so every query represents an email address.

The query `juliet@example.org` would match the uid `Juliet Capulet <juliet@example.org>`.

[Rfc4880](https://tools.ietf.org/html/rfc4880#section-5.11) however states:
```
By convention, it includes an RFC 2822 [RFC2822] mail name-addr, but there are no restrictions on its content.
```

In my case I have to search for uids that do not contain an email address (for example `xmpp:juliet@example.org`).
As the Rfc4880KeySelectionStrategy encloses all queries with `<>`, I am not able to find those uids.

The solution for me was to overwrite the `publicKeyRingsForUid()` methods behaviour in my own KeySelectionStrategy.

I'm reporting this though, since I'm not sure if it is an error in your implementation or not.
Other than that, I found your library very pleasing to use, thank you for that :)

**Comment by neuhalje** [comment]
This is a good find! I'll look into this!

**Comment by vanitasvitae** [comment]
Maybe the Rfc4880KeySelectionStrategy should be named EmailKeySelectionStrategy instead.

**Comment by nmatt** [comment]
I ran into the same problem; it would be nice if this could be fixed.

**Comment by vanitasvitae** [comment]
@nmatt as a workaround you can implement your own `KeySelectionStrategy` ;)

**Comment by nmatt** [comment]
@vanitasvitae: That’s what we’ve done, but it would still be nice if it weren’t necessary, and it’s the only open bug issue. ;)

**Comment by vanitasvitae** [comment]
@nmatt create a PR ;)

**Comment by Alexey1Gavrilov** [comment]
Same problem

**Comment by neuhalje** [comment]
I‘ll Look into this.

**Comment by neuhalje** [comment]
@vanitasvitae / @nmatt / @Alexey1Gavrilov  : Care to review pr #25 ?

**Comment by vanitasvitae** [comment]
@neuhalje haven't looked too deep into it, but on first glance it looks promising :)

**Comment by neuhalje** [comment]
I pushed the fix to master. I'll release a new version tomorrow.

**Comment by neuhalje** [comment]
I just published [2.1.1](https://github.com/neuhalje/bouncy-gpg/releases/tag/release%2Fv2.1.1).

To enable this new feature call "selectUidByAnyUidPart()":

```java
   final InputStream plainIS = BouncyGPG.decryptAndVerifyStream()
        .withConfig(Configs.keyringConfigFromFilesForRecipient())
        .selectUidByAnyUidPart()  // << here
        .andRequireSignatureFromAllKeys("Sven Sender")
        ...
```

**Comment by nmatt** [comment]
@neuhalje Thanks for the quick fix, I can confirm that it solves our use case.
